modules/cron2/Classes/Cron2.php

```diff
@@ -4,48 +4,109 @@
 
 class Cron2
 {
+
+    const LS_CRON_MAX_ERRORS = 3; //Maximum amount of errors for a job
+
     /**
      * @param int $jobid
      * @return bool
      */
     public function Run($jobid)
     {
-        global $db, $func, $config;
+        global $database, $func, $config;
 
         if (!$jobid) {
             return false;
         }
 
-        $row = $db->qry_first("SELECT name, type, `function` FROM %prefix%cron WHERE jobid = %int%", $jobid);
+        $row = $database->queryWithOnlyFirstRow(
+            "SELECT name, type, `function`, error_runs
+            FROM %prefix%cron
+            WHERE jobid = ?",
+            [$jobid]
+        );
+
         if ($row != false) {
-            if ($row['type'] == 'sql') {
-                $sql = str_replace('%prefix%', $config['database']['prefix'], $row['function']);
-                $db->qry('%plain%', $func->AllowHTML($sql));
-            } elseif ($row['type'] == "php") {
-                require_once 'ext_scripts/'.$row['function'];
+            try {
+                $runtime = microtime(true);
```

Comment:
One of the false-hood believes of programmers is that time always runs forward. For time measurement, we should use monotonic clocks (via https://www.php.net/manual/en/function.hrtime.php) and not the UNIX timestamp. Time can go backwards - BUt we want to measure the duration.
See https://blog.codeminer42.com/the-monotonic-clock-and-why-you-should-care-about-it/

Reply:
This was less an assumption of getting a good and exact measurement, but more a _good enough_ time for rough display purposes.
So while it is fit for purpose as-is it is not a big deal to rework to use `hrtime`